mllib/src/main/scala/org/apache/spark/mllib/evaluation/RegressionMetrics.scala

```diff
@@ -31,7 +31,8 @@ import org.apache.spark.sql.DataFrame
  * @param predictionAndObservations an RDD of (prediction, observation) pairs.
  */
 @Experimental
-class RegressionMetrics(predictionAndObservations: RDD[(Double, Double)]) extends Logging {
+class RegressionMetrics(predictionAndObservations: RDD[(Double, Double)])
+  extends Logging with Serializable {
```

Comment:
I think the serialization errors are from LinearRegressionModel being used in closures (for prediction).  That should be fixable by having trainingResults be transient.  We'll have to add doc to the getter/setter for it to warn the user about using those only on the driver.

I'd strongly prefer lazy evaluation.